### PR TITLE
nixos: Add more ssh-keygen params

### PR DESCRIPTION
**Pull request by rvolosatovs**
###### Motivation for this change
Some options, that user may want to be able to specify can now be specified for the key generation procedure.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/networking/ssh/sshd.nix [hunk 1]
```diff
@@ -195,8 +195,8 @@ in
       hostKeys = mkOption {
         type = types.listOf types.attrs;
         default =
-          [ { type = "rsa"; bits = 4096; path = "/etc/ssh/ssh_host_rsa_key"; }
-            { type = "ed25519"; path = "/etc/ssh/ssh_host_ed25519_key"; }
+          [ { type = "rsa"; bits = 4096; path = "/etc/ssh/ssh_host_rsa_key"; rounds = 100; openSSHFormat = true; }
+            { type = "ed25519"; path = "/etc/ssh/ssh_host_ed25519_key"; rounds = 100; comment = "key comment"; }
```
Comment (Mic92) [inline nixos/modules/services/networking/ssh/sshd.nix]: Do we need these default values or should they go to examples?

Comment (rvolosatovs) [reply nixos/modules/services/networking/ssh/sshd.nix]: Indeed, these were meant to be examples

Comment (Mic92) [reply nixos/modules/services/networking/ssh/sshd.nix]: Are the parameter chosen by sshd unreasonable low?

Comment (rvolosatovs) [reply nixos/modules/services/networking/ssh/sshd.nix]: Not sure about that part, but the parameters are taken from [this](https://stribika.github.io/2015/01/04/secure-secure-shell.html) article, which is referenced in the file numerous times already.

File: nixos/modules/services/networking/ssh/sshd.nix [hunk 2]
```diff
@@ -209,7 +213,7 @@ in
 
       authorizedKeysFiles = mkOption {
         type = types.listOf types.str;
-        default = [];
+        default = [ ".ssh/authorized_keys" ".ssh/authorized_keys2" "/etc/ssh/authorized_keys.d/%u" ];
```
Comment (aneeshusa) [inline nixos/modules/services/networking/ssh/sshd.nix]: See https://github.com/NixOS/nixpkgs/pull/10155 for how this works; any values set in `authorizedKeysFiles` are merged into the default, and I don't think we are likely to change this behavior soon due to backwards compatability, so please undo this change. However, since this makes at least 2 people confused by this, I think a documentation update would be great!

File: nixos/modules/services/networking/ssh/sshd.nix [hunk 3]
```diff
@@ -356,7 +360,14 @@ in
 
                 ${flip concatMapStrings cfg.hostKeys (k: ''
                   if ! [ -f "${k.path}" ]; then
-                      ssh-keygen -t "${k.type}" ${if k ? bits then "-b ${toString k.bits}" else ""} -f "${k.path}" -N ""
+                      ssh-keygen\
```
Comment (aneeshusa) [inline nixos/modules/services/networking/ssh/sshd.nix]: style nit: have a space between the `\` continuation at the end of each line

**Review by fpletz — APPROVED** [review]
